awx/main/scheduler/kubernetes.py

```diff
@@ -0,0 +1,96 @@
+import time
+
+from django.conf import settings
+from kubernetes.client import Configuration as kubeconfig
+from kubernetes.client import CoreV1Api as v1_kube_api
+from kubernetes.client import ApiClient as kube_api_client
+
+# AWX
+from awx.main.utils.common import parse_yaml_or_json
+from awx.main.utils import (
+    create_temporary_fifo,
+)
+
+
+class PodManager(object):
+    credential = None
+    kube_api = None
+
+    def __init__(self, task):
+        self.task = task
+        self.credential = task.instance_group.credential
+        self.kube_api = self._kube_api_connection(self.credential)
+        self.pod_name = f"job-{task.id}"
+        self.pod_definition = self._pod_definition()
+
+    def deploy_pod(self):
+        if not self.credential.kubernetes:
+            raise RuntimeError
+
+        self.kube_api.create_namespaced_pod(body=self.pod_definition,
+                                            namespace='awx')
+
+        while True:
+            pod = self.kube_api.read_namespaced_pod(name=self.pod_name,
+                                                    namespace=self.namespace)
+            if pod.status.phase != 'Pending':
+                break
+            time.sleep(1)
+
+        return pod
+
+    @property
+    def namespace(self):
+        return self.pod_definition['metadata']['namespace']
+
+    def _kube_api_connection(self, credential):
+        configuration = kubeconfig()
+        configuration.api_key["authorization"] = credential.get_input('bearer_token')
+        configuration.api_key_prefix['authorization'] = 'Bearer'
+        configuration.host = credential.get_input('host')
+
+        if credential.get_input('verify_ssl'):
+            ca_cert_data = credential.get_input('ssl_ca_cert')
+            configuration.ssl_ca_cert = create_temporary_fifo(ca_cert_data.encode())
```

Comment:
Hot :+1:

Reply:
Yep, this is why I had to use that pesky tempfile on the execution side! Bleh.